### PR TITLE
Added language specification

### PR DESCRIPTION
**Pull request by EmanuelPutura**

### CODE REVIEW

File: language-specification/lexic.txt [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+a. Upper (A-Z) and lower case letters (a-z) of the English alphabet
+b. Underline character '_';
+c. Decimal digits (0-9);
+
+Lexic:
+
+a. Special symbols, representing:
+- operators + - * / % = < <= == > >= != !, and, or
+- separators () [] {}  : ; space ,
+- reserved words: let, int, string, char, array, const, input, output, check, else, loop
```
Comment (HoriaRaduRusu) [inline language-specification/lexic.txt]: while is missing

File: language-specification/syntax.in [hunk 2]
```diff
@@ -0,0 +1,50 @@
+2. Syntax:
+- The words - predefined tokens are specified between " and ". Sintactical rules: (file Syntax.in):
+
+program := "${" {decllist | stmtlist} "}"
+
+decllist := declaration {declaration}
+
+declaration := noconstdeclaration | "const" noconstdeclaration
+
+noconstdeclaration := "let" IDENTIFIER ((":" type1) | ("=" expression) | ("=" arraydecl)) ";"
```
Comment (HoriaRaduRusu) [inline language-specification/syntax.in]: type1 isn't defined anywhere

File: language-specification/syntax.in [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+factor := "(" expression ")" | IDENTIFIER | constant
+
+constant := number | character | string
+
+iostmt := (("input" | "output") "(" IDENTIFIER ")" ";") | ("output" "(" (string | number) ")" ";")
+
+structstmt := cmpdstmt | ifstmt | whilestmt | forstmp
+
+ifstmt := "check" "(" condition ")" "{" stmt "}" ["else check" condition "{" stmt "}"] ["else" "{" stmt "}"]
```
Comment (HoriaRaduRusu) [inline language-specification/syntax.in]: Missing parantheses around the second condition

File: language-specification/lexic.txt [hunk 4]
```diff
@@ -0,0 +1,31 @@
+Alphabet:
+
+a. Upper (A-Z) and lower case letters (a-z) of the English alphabet
+b. Underline character '_';
+c. Decimal digits (0-9);
+
+Lexic:
+
+a. Special symbols, representing:
+- operators + - * / % = < <= == > >= != !, and, or
```
Comment (HoriaRaduRusu) [inline language-specification/lexic.txt:10]: You also use the ++ operator in P3, which isn't defined here

File: language-specification/token.in [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+and
+or
+check
+else
+else check
+loop
+while
+input
+output
++
```
Comment (HoriaRaduRusu) [inline language-specification/token.in:18]: ++ is missing

**Review by HoriaRaduRusu — CHANGES_REQUESTED** [review]

File: language-specification/lexic.txt (hunk 1 above)
Comment (EmanuelPutura) [reply language-specification/lexic.txt]: Done

File: language-specification/syntax.in (hunk 2 above)
Comment (EmanuelPutura) [reply language-specification/syntax.in]: Thanks, the change is done now.

File: language-specification/syntax.in (hunk 3 above)
Comment (EmanuelPutura) [reply language-specification/syntax.in]: Fixed the typo

File: language-specification/token.in (hunk 5 above)
Comment (EmanuelPutura) [reply language-specification/token.in:18]: Done!

File: language-specification/lexic.txt (hunk 4 above)
Comment (EmanuelPutura) [reply language-specification/lexic.txt:10]: Done!

**Review by HoriaRaduRusu — APPROVED** [review]

**Pull request merged**